test/parser/LiteralOrIriTest.cpp

```diff
@@ -278,3 +278,45 @@ TEST(LiteralOrIri, Hashing) {
   ad_utility::HashSet<LiteralOrIri> set{lit, iri};
   EXPECT_THAT(set, ::testing::UnorderedElementsAre(lit, iri));
 }
+
+// _______________________________________________________________________
+TEST(LiteralTest, SetSubstr) {
+  LiteralOrIri literal = LiteralOrIri::literalWithoutQuotes(
+      "Hello World!",
+      Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>"));
+  literal.getLiteral().setSubstr(0, 5, 12);
+  EXPECT_THAT("Hello", asStringViewUnsafe(literal.getContent()));
+  EXPECT_THAT("http://www.w3.org/2001/XMLSchema#string",
+              asStringViewUnsafe(literal.getDatatype()));
+
+  literal = LiteralOrIri::literalWithoutQuotes(
+      "Hello World!",
+      Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>"));
+  literal.getLiteral().setSubstr(6, 5, 12);
+  EXPECT_THAT("World", asStringViewUnsafe(literal.getContent()));
+  EXPECT_THAT("http://www.w3.org/2001/XMLSchema#string",
+              asStringViewUnsafe(literal.getDatatype()));
+
+  literal = LiteralOrIri::literalWithoutQuotes("Äpfel");
```

Comment:
Add a comment here, that the substring works on the byte level, not the UTF-8 level.